src/main/java/de/tum/in/www1/artemis/repository/BuildPlanRepository.java

```diff
@@ -1,12 +1,34 @@
 package de.tum.in.www1.artemis.repository;
 
+import static org.springframework.data.jpa.repository.EntityGraph.EntityGraphType.LOAD;
+
 import java.util.Optional;
 
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 import de.tum.in.www1.artemis.domain.BuildPlan;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
 
 public interface BuildPlanRepository extends JpaRepository<BuildPlan, Long> {
 
+    @Query("""
+            SELECT buildPlan
+            FROM BuildPlan buildPlan
+                INNER JOIN FETCH buildPlan.programmingExercises programmingExercises
+            WHERE programmingExercises.id = :exerciseId
+            """)
+    Optional<BuildPlan> findByProgrammingExercises_IdWithProgrammingExercises(@Param("exerciseId") long exerciseId);
```

Comment:
I do not really like the name now, but this tries to stick mostly with how we name other fetch operations in `@Query`s.